lib/posthog/feature_flags.rb

```diff
@@ -346,7 +403,13 @@ def _load_feature_flags()
       if !res.key?(:flags)
         logger.error "Failed to load feature flags: #{res}"
       else
-        @feature_flags = res[:flags] || []        
+        @feature_flags = res[:flags] || []
```

Comment:
Isn't that fine on successful feature flags load? 

Also wouldn't this be broken on posthog python then too?? 😱 https://github.com/PostHog/posthog-python/blob/a9d7bf3e0b8bdd9c8b1037ad95b106d4cce15084/posthog/client.py#L378

Reply:
nope, that's precisely the problem, we lose flags in between flag loads.

It's not a problem in python because `get()` raises an error and doesn't update existing flags.

This comment was for Line 401, not the current line, but since it's too far away I can't comment on that line :P

Reply:
But if the decide returns something without the flags key (an error), it's handled with the logger so it doesn't update existing flags right?

Reply:
unless I'm misreading something, line 401 clears flags, so if local evaluation errors out, we have empty flags, instead of the previous flags, which is not great.

Instead, we should've kept the old flags.

Reply:
Are you talking about this line? `@feature_flags.clear`

Reply:
yes, the line 401 :P

Reply:
<img width="496" alt="Screen Shot 2023-02-06 at 2 41 28 PM" src="https://user-images.githubusercontent.com/25164963/217068944-a3117682-ec7f-4264-815b-1f7e935e079c.png">

Reply:
ban implicit statements 😤

Reply:
oh noooo 😂 , in my defence, it's line 401 in the PR, which is all I have access to, not your ide lol.

Smh